bundles/core/src/main/java/com/adobe/cq/email/core/components/internal/services/EmailPathProcessor.java

```diff
@@ -0,0 +1,167 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.email.core.components.internal.services;
+
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.scripting.SlingBindings;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.adobe.cq.email.core.components.internal.models.EmailPageImpl;
+import com.adobe.cq.wcm.core.components.services.link.PathProcessor;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.day.cq.wcm.api.PageManagerFactory;
+import com.day.cq.wcm.scripting.WCMBindingsConstants;
+
+/**
+ * This implementation of a {@link PathProcessor} applies a masking Adobe Campaign expressions contained in the path.
+ */
+@Component(
+    service = { EmailPathProcessor.class, PathProcessor.class },
+    property = Constants.SERVICE_RANKING + ":Integer=" + (Integer.MIN_VALUE + 1)
+)
+public class EmailPathProcessor implements PathProcessor {
```

Comment:
As discussed offline the functionality of this `PathProcessor` may be moved to the Core WCM Components `DefaultPathProcessor`.

The implementation masks Adobe Campaign expressions in urls with a random string, then delegates to the `DefaultPathProcessor` and unmasks the returned string afterwards again using the previously created placeholders. This is in particular necessary as the [`DefaultPathProcessor#sanitize()`](https://github.com/adobe/aem-core-wcm-components/blob/core.wcm.components.reactor-2.20.8/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/DefaultPathProcessor.java#L126-L131) method will break urls that are in the form of `<%= targetData.link %>`

Reply:
I moved the logic to the new `LinkBuilderImpl` in adobe/aem-core-wcm-components#2274.